vscode/src/test/suite/extension.test.ts

```diff
@@ -201,15 +208,68 @@ async function testtypedefs(docUri: vscode.Uri) {
     'vscode.executeTypeDefinitionProvider',
     docUri,
     pos1,
-  )) as vscode.Definition[];
-  const loc1 = actualtypedef1[0] as vscode.Location;
+  )) as vscode.Location[];
+  const loc1 = actualtypedef1[0];
   assert.strictEqual(loc1.range.start.line, 7);
   assert.strictEqual(loc1.range.start.character, 4);
   assert.strictEqual(loc1.range.end.line, 21);
   assert.strictEqual(loc1.range.end.character, 5);
   assert.strictEqual(loc1.uri.path, docUri.path);
 }
 
+async function testdecls(docUri: vscode.Uri) {
+  await activate(docUri);
+
+  const pos0 = new vscode.Position(6, 14);
+  const actualdecl0 = (await vscode.commands.executeCommand(
+    'vscode.executeDeclarationProvider',
+    docUri,
+    pos0,
+  )) as vscode.Location[];
+  assert.strictEqual(actualdecl0.length, 2);
```

Comment:
Why does this return two locations, and not the function `foo` in `a` and `b3`?

Reply:
The requirement was to return the overridden functions that belonged to immediate parent contracts. 

So, in the following scenario where `A -> B` means `A` extends `B`, i.e, `A` is a child of `B`
```
     A -> B -> C 
```
when `goto-declaration` is invoked on a function in `A`, only the overridden function in `B` will be returned and not the one in `C`.  

If there are multiple immediate parents, then the overridden functions from all the immediate parents are returned.

Reply:
```
 function foo() internal override(b1, b2, b3) returns (uint64) {
```
This overrides three functions in three contracts.